### PR TITLE
8179503: Java should support GET OCSP calls

### PR DESCRIPTION
**Pull request by apavlyutkin**
Hello! I'd like to backport

8179503: Java should support GET OCSP calls (dependency)

to jdk11u-dev. This one is required as a dependency for 

8274471: Add support for RSASSA-PSS in OCSP Response

The following changes were done to apply original patch:

src/java.base/share/classes/sun/security/provider/certpath/OCSP.java

- resolved baseline conflict taking place due to absent revocation checking code

The rest of the code applied without changes

Verification/regression (20.04/amd64): jdk_security

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8179503](https://bugs.openjdk.org/browse/JDK-8179503): Java should support GET OCSP calls

### Reviewers
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk11u-dev.git pull/847/head:pull/847` \
`$ git checkout pull/847`

Update a local copy of the PR: \
`$ git checkout pull/847` \
`$ git pull https://git.openjdk.org/jdk11u-dev.git pull/847/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 847`

View PR using the GUI difftool: \
`$ git pr show -t 847`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk11u-dev/pull/847.diff">https://git.openjdk.org/jdk11u-dev/pull/847.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk11u-dev/pull/847#issuecomment-1056851985)

### CODE REVIEW

**Comment by apavlyutkin** [comment]
@TheRealMDoerr @alexeybakhtin I splitted https://github.com/openjdk/jdk11u-dev/pull/788, this is the 1st part for 8179503

File: src/java.base/share/classes/sun/security/provider/certpath/OCSP.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2009, 2017, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2009, 2022, Oracle and/or its affiliates. All rights reserved.
```
Comment (TheRealMDoerr) [inline src/java.base/share/classes/sun/security/provider/certpath/OCSP.java:2]: Copyright year update not needed for pure backports when the original change doesn't touch it. May cause merge conflicts. Feel free to revert.

**Review by TheRealMDoerr — APPROVED** [review]
Looks good.

**Comment by alexeybakhtin** [comment]
Looks good.

**Comment by GoeLin** [comment]
Please configure Pre-submit tests.

**Comment by apavlyutkin** [comment]
@GoeLin I've configured pre-submit tests, but they constantly fail on cross-compilation for different target architectures due to absense of required packages. Is that my fault, should I do something else?

**Comment by TheRealMDoerr** [comment]
A lot of pre-submit tests have failed for other PRs, too. I don't think it's your fault. This change is not platform dependent, so I think your tests are fine.

**Comment by GoeLin** [comment]
Thanks for the tests, the failures are unrelated.

**Comment by apavlyutkin** [comment]
jdk11u-fix-no

**Comment by phax** [comment]
This issue causes a regression, if the OCSP server is not returning the `Content-Length` header which at least happens now and then on a DigiCert server. In that case the `contentLength` is set to `Integer.MAX_VALUE` and `IOUtils.readExactlyNBytes` fails with an `EOFException` because it can`t read `Integer.MAX_VALUE` bytes

See also https://github.com/phax/phase4/issues/124 for my original analysis on the Java 17 problem

**Comment by apavlyutkin** [comment]
> This issue causes a regression, if the OCSP server is not returning the `Content-Length` header which at least happens now and then on a DigiCert server. In that case the `contentLength` is set to `Integer.MAX_VALUE` and `IOUtils.readExactlyNBytes` fails with an `EOFException` because it can't read `Integer.MAX_VALUE` bytes
> 
> See also [phax/phase4#124](https://github.com/phax/phase4/issues/124) for my original analysis on the Java 17 problem

Thanks a lot, Philip! I will keep it on mind.

**Comment by apavlyutkin** [comment]
@RealCLanger 

Hi, Chris! As I see you've marked dependent [JDK-8274471](https://bugs.openjdk.org/browse/JDK-8274471) with `jdk11u-todo`. Does it mean I should keep working on this and create depended PR(s)? Previously this one was rejected with `jdk11u-fix-request-no` as inappropriate for jdk11

**Comment by RealCLanger** [comment]
> This issue causes a regression, if the OCSP server is not returning the `Content-Length` header which at least happens now and then on a DigiCert server. In that case the `contentLength` is set to `Integer.MAX_VALUE` and `IOUtils.readExactlyNBytes` fails with an `EOFException` because it can't read `Integer.MAX_VALUE` bytes
> 
> See also [phax/phase4#124](https://github.com/phax/phase4/issues/124) for my original analysis on the Java 17 problem

This is recorded in JBS now as https://bugs.openjdk.org/browse/JDK-8308255

**Comment by RealCLanger** [comment]
> @RealCLanger
> 
> Hi, Chris! As I see you've marked dependent [JDK-8274471](https://bugs.openjdk.org/browse/JDK-8274471) with `jdk11u-todo`. Does it mean I should keep working on this and create depended PR(s)? Previously this one was rejected with `jdk11u-fix-request-no` as inappropriate for jdk11

Hi @apavlyutkin,
I have removed jdk11u-fix-no from [JDK-8179503](https://bugs.openjdk.org/browse/JDK-8179503) now, as I think it was backported for Oracle as well (although the backport is not public) and there is justification for backporting this change as well as [JDK-8274471](https://bugs.openjdk.org/browse/JDK-8274471).
You need not care about the jdk11u-todo label. It is just some flagging that helps us keep the JBS filters for open backports lean. You could remove the label once a backport to 11u was merged. But you can also keep the label around for us to clean up.

So, please go ahead with your efforts to backport these two items. As for this PR, I think a merge with master would clean it up and also trigger GHA, hoping for all green results.

**Review by phohensee — CHANGES_REQUESTED** [review]
@apavlyutkin, there are 494 changed files in the PR vs 3 in the original commit. Looks like a merge issue.

**Review by phohensee — CHANGES_REQUESTED** [review]
Somehow the last merge changed 494 files, please fix.

**Comment by apavlyutkin** [comment]
> Somehow the last merge changed 494 files, please fix.

The request was initiated Mar 2, 2022. This is just an upmerge

**Comment by TheRealMDoerr** [comment]
It may be correct. We can see the actual patch here: https://patch-diff.githubusercontent.com/raw/openjdk/jdk11u-dev/pull/847.patch
The GitHub PR view is broken. We can't see what is being changed. Maybe it's possible to fix it by changing what exactly is supposed to get diffed.

**Pull request closed without merging**